core/templates/dev/head/services/StateRulesStatsService.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for calculating the statistics of a particular state.
+ */
+
+oppia.factory('StateRulesStatsService', [
+  '$http', '$injector', 'AngularNameService', 'AnswerClassificationService',
+  'ExplorationContextService', 'UrlInterpolationService',
+  function(
+      $http, $injector, AngularNameService, AnswerClassificationService,
+      ExplorationContextService, UrlInterpolationService) {
+    return {
+      /**
+       * TODO(brianrodri): Consider moving this into a visualization domain
+       * object.
+       *
+       * @param {Object!} state
+       * @return {Boolean} whether given state has an implementation for
+       *     displaying the issues overview tab in the State Editor.
+       */
+      stateSupportsIssuesOverview: function(state) {
+        return state.interaction.id === 'TextInput';
+      },
+
+      /**
+       * Returns a promise which will provide details of the given state's
+       * answer-statistics.
+       *
+       * @param {Object!} state
+       * @param {String?} testOnlyExplorationId used to enforce a specific id
```

Comment:
This is still a bit clumsy though. Can't you just mock the relevant explorationContextService method when testing? That would be cleaner.

Reply:
Done, yes that does look much better :)